src/api/app/models/token.rb

```diff
@@ -1,10 +1,14 @@
 class Token < ApplicationRecord
-  belongs_to :user, foreign_key: 'user_id', inverse_of: :service_tokens
+  belongs_to :user, foreign_key: 'user_id'
   belongs_to :package, inverse_of: :tokens
 
   has_secure_token :string
 
   validates :user, presence: true
+
+  def token_name
```

Comment:
why don't you just define it as an instance method in the subclass? :confused:

Reply:
I am not sure if it clear what I meant. You could define for example in `rebuild.rb`:

```
def token_name
  'rebuild'
end
```

without the `self`

Reply:
^^

Reply:
his could be an instance method of the subclass. Any reason to do it like this needing to do `self.class.token_name`?

Reply:
this  is also not addressed...

Reply:
IIRC, this could also be a delegate to the class, but I think that's more of a nitpick

Reply:
we didn't want to change it before we have specs for that.

Reply:
I'm not sure what would be the difference here, but because it was something that Adrian did, and I don't see why we should change it now, I would keep it (unless there is a strong argument to change it)